sound/soc/sof/sof-audio.h

```diff
@@ -290,6 +290,9 @@ struct snd_sof_pcm_stream {
 	 * active or not while suspending the stream
 	 */
 	bool suspend_ignored;
+
+	/* calculate delay for different ipc version */
+	void *time_info;
```

Comment:
The "void *" does not look good here. @ujfalusi @ranj063 do we have any good examples how to wrap structs that may be different for IPC3 and IPC4 for sof-audio?

Reply:
Currently we deal with ipc4_fw_data in the same way
````c
struct snd_sof_dev {
                ...
                void *private;			/* core does not touch this */
}

struct sof_ipc4_fw_data *ipc4_data = sdev->private;
````

Reply:
I think we do need to have this indirection, but I would go with a generic name, like `ipc_priv_data` or `private` and document that it is used by IPC implementation and core does not touch it.